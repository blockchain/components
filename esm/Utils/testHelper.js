import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}