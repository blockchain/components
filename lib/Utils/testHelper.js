"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}