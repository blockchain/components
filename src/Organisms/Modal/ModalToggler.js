// @flow strict
import React from 'react'
import { FasTimes } from '../../Atoms/Icons'

export type PropsType = {}

const ModalToggler = ({ ...props }: PropsType) => (
  <FasTimes size="28px" {...props} />
)

export default ModalToggler
