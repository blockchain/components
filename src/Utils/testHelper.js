import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
