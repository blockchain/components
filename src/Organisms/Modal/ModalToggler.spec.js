import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import ModalToggler from './ModalToggler'

describe('ModalToggler', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<ModalToggler />)

    expect(tree.find(ModalToggler)).toMatchSnapshot()
  })
})
