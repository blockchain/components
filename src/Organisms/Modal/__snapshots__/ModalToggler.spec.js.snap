// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalToggler should render correctly 1`] = `
.c0 {
  fill: #98A1B2;
  cursor: default;
}

.c0:hover {
  fill: #98A1B2;
}

<ModalToggler>
  <FasTimes
    selectable={false}
    size="28px"
  >
    <Styled(SvgFasTimes)
      height="28px"
      selectable={0}
      width="28px"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-cMjzQU",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
    ",
                "
  fill: ",
                [Function],
                ";
  cursor: ",
                [Function],
                ";

  &:hover {
    fill: ",
                [Function],
                ";
  }
",
                ";
  ",
              ],
            },
            "displayName": "Styled(SvgFasTimes)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-cMjzQU",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        height="28px"
        selectable={0}
        width="28px"
      >
        <SvgFasTimes
          className="c0"
          height="28px"
          selectable={0}
          width="28px"
        >
          <svg
            className="c0"
            height="28px"
            selectable={0}
            viewBox="0 0 352 512"
            width="28px"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
        </SvgFasTimes>
      </StyledComponent>
    </Styled(SvgFasTimes)>
  </FasTimes>
</ModalToggler>
`;
